src/btree.h

```diff
@@ -341,26 +346,27 @@ btree_pack_req_init(btree_pack_req  *req,
    req->hash       = hash;
    req->seed       = seed;
    if (hash != NULL && max_tuples > 0) {
-      req->fingerprint_arr =
-         TYPED_ARRAY_ZALLOC(hid, req->fingerprint_arr, max_tuples);
+
+      fingerprint_init(&req->fingerprint, hid, max_tuples); // Allocates memory
 
       // When we run with shared-memory configured, we expect that it is sized
       // big-enough to not get OOMs from here. Hence, only a debug_assert().
-      debug_assert(req->fingerprint_arr,
+      debug_assert(!fingerprint_is_empty(&req->fingerprint),
                    "Unable to allocate memory for %lu tuples",
                    max_tuples);
-      if (!req->fingerprint_arr) {
+      if (fingerprint_is_empty(&req->fingerprint)) {
```

Comment:
You can no longer check for NULL array ptr, to detect OOM. You must consult the `is_empty()` method to figure out if there is memory or not.